### PR TITLE
[0.64] Secure SampleAppCpp Certificate

### PR DESCRIPTION
**Pull request by chiaramooney**
**_Why_**
SampleAppCpp certificate has expired; Backport #7880 to 0.64 to allow for pipeline to return to a successful build state.

**_What_**
Removed SampleAppCpp Certificate. Adjusted pipeline code to retrieve certificate from secure storage. 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/8436)

### CODE REVIEW

**Comment by NickGerleman** [comment]
@chiaramooney it looks like hitting a YAML parse error. The difference between PR and CI envrionments was added in 0.65, so there might need to be some changes to this PR. See https://dev.azure.com/ms/react-native-windows/_build/results?buildId=212538&view=results

image


Usually when a run fails to start at all, this is the cause. You can find the results if you go to the Azure Pipelines page manually.

File: .ado/windows-vs-pr.yml [hunk 1]
```diff
@@ -359,19 +359,16 @@ jobs:
...
-          workingDirectory: packages/microsoft-reactnative-sampleapps
-        condition: and(succeeded(), eq(variables['BuildConfiguration'], 'Debug'))
-
-      - task: CmdLine@2
-        displayName: run-windows (Release)
-        inputs:
-          script: yarn windows --no-packager --no-launch $(NoDeployOption) --arch $(BuildPlatform) --logging --buildLogDirectory $(BuildLogDirectory) --release --msbuildprops BaseIntDir=$(BaseIntDir)
+      - template: ../templates/run-windows-with-certificates.yml
+        parameters:
+          buildEnvironment: ${{ parameters.BuildEnvironment }}
```
Comment (jonthysell) [inline .ado/windows-vs-pr.yml:364]:
```suggestion
          buildEnvironment: 'PullRequest'
```

**Review by jonthysell — APPROVED** [review]

File: .ado/windows-vs-pr.yml [hunk 2]
```diff
@@ -359,19 +359,16 @@ jobs:
...
-        inputs:
-          script: yarn windows --no-packager --no-launch $(NoDeployOption) --arch $(BuildPlatform) --logging --buildLogDirectory $(BuildLogDirectory) --msbuildprops BaseIntDir=$(BaseIntDir)
-          workingDirectory: packages/microsoft-reactnative-sampleapps
-        condition: and(succeeded(), eq(variables['BuildConfiguration'], 'Debug'))
-
-      - task: CmdLine@2
-        displayName: run-windows (Release)
-        inputs:
-          script: yarn windows --no-packager --no-launch $(NoDeployOption) --arch $(BuildPlatform) --logging --buildLogDirectory $(BuildLogDirectory) --release --msbuildprops BaseIntDir=$(BaseIntDir)
+      - template: ../templates/run-windows-with-certificates.yml
```
Comment (jonthysell) [inline .ado/windows-vs-pr.yml:362]:
```suggestion
      - template: templates/run-windows-with-certificates.yml
```

**Comment by jonthysell** [comment]
Merging this in as the PR is now running correctly, so future 0.64 backports will have working PR pipeline.

**Pull request merged**